### PR TITLE
Combine brave-extension with brave-core

### PR DESCRIPTION
**Pull request by bbondy**
Fix [brave/brave-browser#1506](https://github.com/brave/brave-browser/issues/1506)
Fix [brave/brave-browser#3399](https://github.com/brave/brave-browser/issues/3399)
Fix [brave/brave-browser#1555](https://github.com/brave/brave-browser/issues/1555)

# This is part 2 of 2 PRs on brave/brave-core.

Various fixes for removing duplicated code and having a separate repo for brave-extension
This is much easier to review commit by commit.

Part 1 of 2 can be found here:
https://github.com/brave/brave-core/pull/1754

brave/brave-browser branch for this is here:
https://github.com/brave/brave-browser/pull/3446

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Windows
  - [x] macOS
  - [x] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [x] Windows
  - [x] macOS
  - [x] Linux
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: components/test/testSetup.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ import * as EnzymeAdapter from 'enzyme-adapter-react-16'
 // since we are not testing action creators ATM it seemed ok to have it mocked
 // to unblock other work depending on this test refactor.
 // TODO: @cezaraugusto investigate
-jest.mock('../brave_extension/extension/brave_extension/background/actions/shieldsPanelActions.ts')
+jest.mock('../brave_extension/extension/brave_extension/background/actions/shieldsPanelActions')
```
Comment (cezaraugusto) [inline components/test/testSetup.ts]: turns out this was related to the circular dependency bug. I did `npm run test-unit` and all tests passed without this mock so it's safe to remove

File: package.json [hunk 2]
```diff
@@ -294,14 +296,15 @@
     "react-dom": "^16.3.0",
     "react-redux": "^5.0.6",
     "redux": "^4.0.0",
+    "sinon": "^7.2.4",
```
Comment (cezaraugusto) [inline package.json]: interesting that TS doesn't complain about missing `@types/sinon` anymore

File: components/test/testSetup.ts (hunk 1 above)
Comment (bbondy) [reply components/test/testSetup.ts]: Fails for me if I remove this, but passes with it.  So I'll keep it for now and we can figure it out later.

File: package.json (hunk 2 above)
Comment (bbondy) [reply package.json]: 🤷‍♂️

**Pull request merged**